cabal-install/Distribution/Client/Dependency/Modular/Cycles.hs

```diff
@@ -48,26 +47,16 @@ detectCyclesPhase = (`runReader` Set.empty) .  cata go
 -- as the full conflict set containing all decisions that led to that 'Done'
 -- node, check if the solution is cyclic. If it is, return the conflict set
 -- containing all decisions that could potentially break the cycle.
-findCycles :: ConflictSet QPN -> RevDepMap -> Maybe (ConflictSet QPN)
-findCycles fullSet revDeps = do
+findCycles :: Map QPN QGoalReason -> RevDepMap -> Maybe (ConflictSet QPN)
+findCycles grs revDeps = do
     guard $ not (null cycles)
```

Comment:
I'd certainly prefer a simple
```
case cycles of
  [] -> Nothing
  (c : _) -> Just $ ...
```
over this combintion of `guard`, `null` and `head`.